File: 4th Semester/Systems for Design and Implementation/C#/Homework1/OnwardTravel/models/Reservation.cs
namespace OnwardTravel.models
{
    internal class Reservation : Entity<Guid>
    {
        public Client Client { get; }
        public Ride Ride { get; }
        public List<ReservedSeat> Seats { get; }

        public Reservation(Guid guid, Client client, Ride ride, params ReservedSeat[] seats)
            : base(guid)
        {
            Client = client;
            Ride = ride;
            Seats = new(seats);
        }
        public Reservation(Client client, Ride ride, params ReservedSeat[] seats)
        {
            Client = client;
            Ride = ride;
            Seats = new(seats);
        }
    }
}

File: 4th Semester/Systems for Design and Implementation/Java/Homework4/OnwardModels/src/main/java/models/Reservation.java
package models;

import java.util.List;
import java.util.UUID;

public class Reservation extends Entity<UUID> {
    private final Client client;
    private final Ride ride;
    private final List<ReservedSeat> seats;

    public Reservation(UUID uuid, Client client, Ride ride, ReservedSeat... seats) {
        super(uuid);
        this.client = client;
        this.ride = ride;
        this.seats = List.of(seats);
    }

    public Reservation(Client client, Ride ride,ReservedSeat... seats) {
        this.client = client;
        this.ride = ride;
        this.seats = List.of(seats);
    }

    public Reservation(UUID uuid, Client client, Ride ride, List<ReservedSeat> seats) {
        super(uuid);
        this.client = client;
        this.ride = ride;
        this.seats = seats;
    }

    public Reservation(Client client, Ride ride, List<ReservedSeat> seats) {
        this.client = client;
        this.ride = ride;
        this.seats = seats;
    }

    public Client getClient() {
        return client;
    }

    public Ride getRide() {
        return ride;
    }
    public List<ReservedSeat> getSeats() {
        return seats;
    }
}
